staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -136,6 +137,41 @@ func (in *Unstructured) DeepCopy() *Unstructured {
 	return out
 }
 
+// GetFieldValue returns the value at the given
+// path if it's a string, else error. E.g.:
+//   u.GetFieldValue("Kind") == "Deployment"
+//   u.GetFieldValue("spec.ports.port") == "8080"
+func (in *Unstructured) GetFieldValue(path string) (string, error) {
```

Comment:
We already have lots of getters and setters here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go :)

Reply:
Dropping this PR in favor of your suggestion.

Given a file `foo.go` containing the type definition for `Foo`, it would never occur to me to look in `helpers.go` for `Foo` getters.  Is this just a historical accident?

Reply:
> Given a file foo.go containing the type definition for Foo, it would never occur to me to look in helpers.go for Foo getters. Is this just a historical accident?

I'm not familiar with the historical context of this, but I guess having it under `util/` or something similar makes more sense. `helpers.go` indeed seems more "internal".